code/game/objects/mail.dm

```diff
@@ -0,0 +1,270 @@
+	. = ..()
+	var/item = pick(possible_contents)
+	new item(src)
+	new /obj/item/stack/spacecash(src, rand(1, 50) * 5)
+	var/list/mind_copy = shuffle(SSticker.minds)
+	for(var/datum/mind/mail_attracted_people in mind_copy)
+		var/turf/T = get_turf(mail_attracted_people.current)
+		if(mail_attracted_people.offstation_role || !ishuman(mail_attracted_people.current) || T.z == 1)
+			continue
+		if(mail_attracted_people.assigned_role in job_list)
+			recipient = mail_attracted_people.current
+			name = "letter to [recipient.real_name]"
+			return
+	if(!admin_spawned)
+		log_debug("Failed to find a new name to assign to [src]!")
+		qdel(src)
+
+/obj/item/envelope/security
+	icon_state = "mail_sec"
+	possible_contents = list(/obj/item/reagent_containers/food/snacks/donut/sprinkles,
+							/obj/item/megaphone,
+							/obj/item/poster/random_official,
+							/obj/item/restraints/handcuffs/pinkcuffs,
+							/obj/item/restraints/legcuffs/bola/energy,
+							/obj/item/reagent_containers/food/drinks/coffee,
+							/obj/item/stock_parts/cell/super,
+							/obj/item/grenade/barrier/dropwall,
+							/obj/item/toy/figure/crew/detective,
+							/obj/item/toy/figure/crew/hos,
+							/obj/item/toy/figure/crew/secofficer)
+	job_list = list("Head of Security", "Security Officer", "Detective", "Warden")
+
+/obj/item/envelope/science
+	icon_state = "mail_sci"
+	possible_contents = list(/obj/item/analyzer,
+							/obj/item/assembly/signaler,
+							/obj/item/slime_extract/grey,
+							/obj/item/clothing/mask/gas,
+							/obj/item/reagent_containers/spray/cleaner,
+							/obj/item/clothing/glasses/regular,
+							/obj/item/stack/ore/diamond, // Jackpot
+							/obj/item/paicard,
+							/obj/item/toy/figure/crew/borg,
+							/obj/item/toy/figure/crew/geneticist,
+							/obj/item/toy/figure/crew/rd,
+							/obj/item/toy/figure/crew/roboticist,
+							/obj/item/toy/figure/crew/scientist)
+	job_list = list("Research Director", "Roboticist", "Geneticist",  "Scientist")
+
+/obj/item/envelope/supply
+	icon_state = "mail_sup"
+	possible_contents = list(/obj/item/reagent_containers/hypospray/autoinjector/survival,
+							/obj/item/reagent_containers/food/drinks/bottle/absinthe/premium,
+							/obj/item/clothing/glasses/meson/gar,
+							/obj/item/stack/marker_beacon/ten,
+							/obj/item/stack/medical/splint,
+							/obj/item/pen/multi/fountain,
+							/obj/item/clothing/mask/cigarette/cigar,
+							/obj/item/stack/wrapping_paper,
+							/obj/item/toy/figure/crew/cargotech,
+							/obj/item/toy/figure/crew/qm,
+							/obj/item/toy/figure/crew/miner)
+	job_list = list("Quartermaster", "Cargo Technician", "Shaft Miner")
+
+/obj/item/envelope/medical
+	icon_state = "mail_med"
+	possible_contents = list(/obj/item/soap,
+							/obj/item/reagent_containers/glass/bottle/morphine,
+							/obj/item/reagent_containers/hypospray/safety,
+							/obj/item/reagent_containers/applicator/brute,
+							/obj/item/reagent_containers/applicator/burn,
```

Comment:
Indentation on these envelopes have yet to be adjusted.

Reply:
Resolved on discord